app/components/Settings/Settings.react.js

```diff
@@ -445,14 +444,19 @@ function mergeProps(stateProps, dispatchProps, ownProps) {
         dispatchConnect = function saveAndConnect() {
             dispatch(Actions.saveConnections(connections[selectedTab], selectedTab))
             .then(json => {
-                dispatch(Actions.connect(json.connectionId));
+                /*
+                 * Do not try connecting because saving of credentials
+                 * will not have happened on error
+                 */
+                if (!json.error) {
+                    dispatch(Actions.connect(json.connectionId));
+                }
             });
             // TODO - If connecting *fails* then we should delete the connection
```

Comment:
we can ⚡️  this comment now since we're not saving unless we have a successful connection